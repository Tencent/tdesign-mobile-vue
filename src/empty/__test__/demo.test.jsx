/**
 * 该文件为由脚本 `npm run test:demo` 自动生成，如需修改，执行脚本命令即可。请勿手写直接修改，否则会被覆盖
 */

import { mount } from '@vue/test-utils';
import baseVue from '@/empty/demos/base.vue';
import buttonEmptyVue from '@/empty/demos/buttonEmpty.vue';
import imageEmptyVue from '@/empty/demos/imageEmpty.vue';
import mobileVue from '@/empty/demos/mobile.vue';

const mapper = {
  baseVue,
  buttonEmptyVue,
  imageEmptyVue,
  mobileVue,
};

describe('Empty', () => {
  Object.keys(mapper).forEach((demoName) => {
    it(`Empty ${demoName} demo works fine`, () => {
      const wrapper = mount(mapper[demoName]);
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
