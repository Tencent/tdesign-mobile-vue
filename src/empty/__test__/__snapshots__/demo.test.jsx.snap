// Vitest Snapshot v1

exports[`Empty > Empty baseVue demo works fine 1`] = `
<div
  class="t-empty"
>
  <div
    class="t-empty__thumb"
  >
    <div
      class="t-empty__icon"
    >
      <svg
        class="t-icon t-icon-info-circle-filled"
        fill="none"
        height="1em"
        viewBox="0 0 24 24"
        width="1em"
      >
        <path
          d="M12 23a11 11 0 100-22 11 11 0 000 22zM11 8.5v-2h2v2h-2zm2 1.5v7.5h-2V10h2z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
  <div
    class="t-empty__description"
  >
    描述文字
  </div>
  <!---->
</div>
`;

exports[`Empty > Empty buttonEmptyVue demo works fine 1`] = `
<div
  class="t-empty"
>
  <div
    class="t-empty__thumb"
  >
    <div
      class="t-empty__icon"
    >
      <svg
        class="t-icon t-icon-info-circle-filled"
        fill="none"
        height="1em"
        viewBox="0 0 24 24"
        width="1em"
      >
        <path
          d="M12 23a11 11 0 100-22 11 11 0 000 22zM11 8.5v-2h2v2h-2zm2 1.5v7.5h-2V10h2z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
  <div
    class="t-empty__description"
  >
    描述文字
  </div>
  <div
    class="t-empty__actions"
  >
    
    <button
      aria-disabled="false"
      class="t-button t-button--size-large t-button--base t-button--primary t-button--rectangle"
      role="button"
      type="button"
    >
      <!---->
      <span
        class="t-button__content"
      >
        
        操作按钮
        
      </span>
      <!---->
    </button>
    
  </div>
</div>
`;

exports[`Empty > Empty imageEmptyVue demo works fine 1`] = `
<div
  class="t-empty"
>
  <div
    class="t-empty__thumb"
  >
    <div
      class="t-image t-image--square"
    >
      <div
        class="t-image__mask"
      >
        <div
          class="t-loading"
          style="font-size: 20px;"
        >
          <div
            class="t-loading__dots"
            style="animation-duration: 800ms; width: 20px; height: 20px;"
          >
            
            <div
              class="t-loading__dot"
              style="animation-duration: 0.8s; animation-delay: 0s;"
            />
            <div
              class="t-loading__dot"
              style="animation-duration: 0.8s; animation-delay: 0.26666666666666666s;"
            />
            <div
              class="t-loading__dot"
              style="animation-duration: 0.8s; animation-delay: 0.5333333333333333s;"
            />
            
          </div>
          <!---->
          <!---->
        </div>
      </div>
      <picture>
        <!--v-if-->
        <img
          alt=""
          class="t-image__img"
          src="https://tdesign.gtimg.com/mobile/demos/empty1.png"
          style="object-fit: fill; object-position: center;"
        />
      </picture>
    </div>
  </div>
  <div
    class="t-empty__description"
  >
    描述文字
  </div>
  <!---->
</div>
`;

exports[`Empty > Empty mobileVue demo works fine 1`] = `
<div
  class="tdesign-mobile-demo"
>
  <h1
    class="title"
  >
    Empty 空状态
  </h1>
  <p
    class="summary"
  >
    用于空状态时的占位提示。
  </p>
  <div
    class="tdesign-mobile-demo-block tdesign-mobile-demo-block_notitle"
  >
    <div
      class="tdesign-mobile-demo-block__header"
    >
      <!--v-if-->
      <p
        class="tdesign-mobile-demo-block__summary tdesign-mobile-demo-block_subtitle"
      >
        图标空状态
      </p>
    </div>
    <div
      class="tdesign-mobile-demo-block__slot"
    >
      
      <div
        class="t-empty"
      >
        <div
          class="t-empty__thumb"
        >
          <div
            class="t-empty__icon"
          >
            <svg
              class="t-icon t-icon-info-circle-filled"
              fill="none"
              height="1em"
              viewBox="0 0 24 24"
              width="1em"
            >
              <path
                d="M12 23a11 11 0 100-22 11 11 0 000 22zM11 8.5v-2h2v2h-2zm2 1.5v7.5h-2V10h2z"
                fill="currentColor"
              />
            </svg>
          </div>
        </div>
        <div
          class="t-empty__description"
        >
          描述文字
        </div>
        <!---->
      </div>
      
    </div>
  </div>
  <div
    class="tdesign-mobile-demo-block tdesign-mobile-demo-block_notitle"
  >
    <div
      class="tdesign-mobile-demo-block__header"
    >
      <!--v-if-->
      <p
        class="tdesign-mobile-demo-block__summary tdesign-mobile-demo-block_subtitle"
      >
        自定义图片空状态
      </p>
    </div>
    <div
      class="tdesign-mobile-demo-block__slot"
    >
      
      <div
        class="t-empty"
      >
        <div
          class="t-empty__thumb"
        >
          <div
            class="t-image t-image--square"
          >
            <div
              class="t-image__mask"
            >
              <div
                class="t-loading"
                style="font-size: 20px;"
              >
                <div
                  class="t-loading__dots"
                  style="animation-duration: 800ms; width: 20px; height: 20px;"
                >
                  
                  <div
                    class="t-loading__dot"
                    style="animation-duration: 0.8s; animation-delay: 0s;"
                  />
                  <div
                    class="t-loading__dot"
                    style="animation-duration: 0.8s; animation-delay: 0.26666666666666666s;"
                  />
                  <div
                    class="t-loading__dot"
                    style="animation-duration: 0.8s; animation-delay: 0.5333333333333333s;"
                  />
                  
                </div>
                <!---->
                <!---->
              </div>
            </div>
            <picture>
              <!--v-if-->
              <img
                alt=""
                class="t-image__img"
                src="https://tdesign.gtimg.com/mobile/demos/empty1.png"
                style="object-fit: fill; object-position: center;"
              />
            </picture>
          </div>
        </div>
        <div
          class="t-empty__description"
        >
          描述文字
        </div>
        <!---->
      </div>
      
    </div>
  </div>
  <div
    class="tdesign-mobile-demo-block tdesign-mobile-demo-block_notitle"
  >
    <div
      class="tdesign-mobile-demo-block__header"
    >
      <!--v-if-->
      <p
        class="tdesign-mobile-demo-block__summary tdesign-mobile-demo-block_subtitle"
      >
        带操作空状态
      </p>
    </div>
    <div
      class="tdesign-mobile-demo-block__slot"
    >
      
      <div
        class="t-empty"
      >
        <div
          class="t-empty__thumb"
        >
          <div
            class="t-empty__icon"
          >
            <svg
              class="t-icon t-icon-info-circle-filled"
              fill="none"
              height="1em"
              viewBox="0 0 24 24"
              width="1em"
            >
              <path
                d="M12 23a11 11 0 100-22 11 11 0 000 22zM11 8.5v-2h2v2h-2zm2 1.5v7.5h-2V10h2z"
                fill="currentColor"
              />
            </svg>
          </div>
        </div>
        <div
          class="t-empty__description"
        >
          描述文字
        </div>
        <div
          class="t-empty__actions"
        >
          
          <button
            aria-disabled="false"
            class="t-button t-button--size-large t-button--base t-button--primary t-button--rectangle"
            role="button"
            type="button"
          >
            <!---->
            <span
              class="t-button__content"
            >
              
              操作按钮
              
            </span>
            <!---->
          </button>
          
        </div>
      </div>
      
    </div>
  </div>
</div>
`;
